addon/components/table.ts

```diff
@@ -1,110 +1,243 @@
+     *
+     */
+    onReorder?: () => any;
+    /**
+     * An action that is triggered when columns are resized.
+     *
+     */
+    onResize?: () => any;
     sortFunction?: (
-        itemA: TableColumn,
-        itemB: TableColumn,
+        itemA: ColumnValue,
+        itemB: ColumnValue,
         sorts: TableSort[],
         compare: <T>(valueA: T, valueB: T, sortEmptyLast: boolean) => number,
         sortEmptyLast: boolean
     ) => number;
     compareFunction?: <T>(valueA: T, valueB: T, sortEmptyLast: boolean) => number;
-    //the selector used by <VerticalCollection> to calculate occulsion rendering
-    //set this to `null` for fixed height/scrollable tables
-    containerSelector: string = 'body';
-    widthConstraint: string = 'lte-container';
-    fillMode: string = 'first-column';
-    fillColumnIndex: number | null = null;
-    resizeMode: string = 'standard';
-    bufferSize: number = 0;
-    estimateRowHeight: number = 30;
-    renderAllRows: boolean = false;
-    constrainColumnsToFit: boolean = true;
-    showEmptyFooter: boolean = false;
+}
 
-    //component state
-    columns: NativeArray<TableColumn> = A();
-    visibleColumns: NativeArray<TableColumn> = A();
-    columnPanPosition: number = 0;
-    rows: any[] = [];
-    sorts: TableSort[] = [];
-    isLoading: boolean = false;
-    hasMoreRows: boolean = false;
-    hasHiddenOverflow: boolean = false;
-    containerWidth: number | null = null;
+class TableComponent extends Component<TableArgs> {
+    @service media!: Media;
 
     //readonly attributes
 
-    /**
-     * CP that returns the default no-results text
-     *
-     * @type {string} defaultNoResultsText
-     * @memberof TableComponent
-     */
-    @readOnly('noResultsText') defaultNoResultsText!: string;
+    //ember-table's resizing must be enabled in order for fill-mode auto column
+    //resizing to work, even if you don't want to allow user-invoked resizing
+    readonly enableResize: boolean = true;
+    readonly elementId = guidFor(this);
 
     /**
-     * CP that returns whether the table rows are clickable
+     * CP that returns a unique data-table ID
      *
-     * @type {boolean} clickableRows
+     * @readonly
+     * @type {string} tableId
      * @memberof TableComponent
      */
-    @or('onRowClick', 'onRowDoubleClick') clickableRows!: boolean;
+    get tableId(): string {
+        return `data-table-${this.elementId}`;
+    }
 
     /**
-     * CP that returns whether the table is empty
+     * CP that returns a CSS style for the table height
      *
-     * @type {boolean} noRows
+     *
+     * @readonly
+     * @type {string} height
      * @memberof TableComponent
      */
-    @empty('rows') noRows!: boolean;
+    @computed('tableHeight')
+    get height(): SafeString {
+        return htmlSafe(this.tableHeight ? `height: ${this.tableHeight};` : '');
+    }
+
+    get resizeDebounce(): number {
+        return this.args.resizeDebounce ?? 250;
+    }
```

Comment:
I'd love if we could eventually turn this pattern into a decorator or something, e.g.
```ts
@argDefault('resizeDebounce') resizeDebounce = 250;
```
maybe something to experiment with if you get some down time 😄